buildpacks/ruby/src/layers/bundle_download_layer.rs

```diff
@@ -1,149 +1,137 @@
+            cmd.args(["--version", &metadata.version.to_string()])
+                .env_clear()
+                .envs(env);
+
+            // Format `gem install --version <version>` without other content for display
+            let short_name = fun_run::display(&mut cmd);
+
+            // Directory where bundler's contents will live
+            cmd.args(["--install-dir", &format!("{}", layer_ref.path().display())]);
+            // Directory where `bundle` executable lives
+            cmd.args(["--bindir", &format!("{}", bin_dir.display())]);
+            cmd.args([
+                "--force",       // Overwrite if it already exists
+                "--no-document", // Don't install ri or rdoc documentation, which takes extra time
+                "--env-shebang", // Start the `bundle` executable with `#! /usr/bin/env ruby`
+            ]);
+
+            let timer = bullet.start_timer(format!("Running {}", style::command(short_name)));
+            cmd.named_output()
+                .map_err(|error| {
+                    fun_run::map_which_problem(error, cmd.mut_cmd(), env.get("PATH").cloned())
+                })
+                .map_err(RubyBuildpackError::GemInstallBundlerCommandError)?;
+            bullet = timer.done();
         }
     }
+    layer_ref.write_env(
+        LayerEnv::new()
+            .chainable_insert(Scope::All, ModificationBehavior::Delimiter, "PATH", ":")
+            .chainable_insert(
+                Scope::All,
+                ModificationBehavior::Prepend,
+                "PATH", // Ensure this path comes before default bundler that ships with ruby, don't rely on the lifecycle
+                bin_dir,
+            )
+            .chainable_insert(Scope::All, ModificationBehavior::Delimiter, "GEM_PATH", ":")
+            .chainable_insert(
+                Scope::All,
+                ModificationBehavior::Prepend,
+                "GEM_PATH", // Bundler is a gem too, allow it to be required
+                gem_path,
+            ),
+    )?;
+    layer_ref.write_metadata(metadata)?;
+
+    Ok((bullet, layer_ref.read_env()?))
 }
 
-// [derive(Debug)]
-enum State {
-    NothingChanged(ResolvedBundlerVersion),
-    BundlerVersionChanged(ResolvedBundlerVersion, ResolvedBundlerVersion),
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub(crate) struct Metadata {
+    pub(crate) version: ResolvedBundlerVersion,
 }
 
-fn cache_state(old: BundleDownloadLayerMetadata, now: BundleDownloadLayerMetadata) -> State {
-    let BundleDownloadLayerMetadata { version } = now; // Ensure all properties are checked
+fn metadata_diff(old: &Metadata, metadata: &Metadata) -> Option<String> {
+    let Metadata { version } = old;
 
-    if old.version == version {
-        State::NothingChanged(version)
+    if version == &metadata.version {
+        None
     } else {
-        State::BundlerVersionChanged(old.version, version)
+        Some(format!(
+            "Bundler version ({old} to {now})",
```

Comment:
Use lower-case (like the "invalid metadata" cause) https://github.com/heroku/buildpacks-ruby/pull/326/files#diff-cd4ee747be3abfb0ab939cdf3087fa781357327a5eabb3d870f73cc4625b7160R38 and [consistent with the output prior to this PR](https://github.com/heroku/buildpacks-ruby/pull/326/files#diff-cd4ee747be3abfb0ab939cdf3087fa781357327a5eabb3d870f73cc4625b7160L125) - unless this is intentional/desired ("Bundler" could work with the edit I made above, wrapping the `cause` in parenthesis). Also added "changed" for context.

```suggestion
            "bundler version changed ({old} to {now})",
```